### ISSUE TITLE
Matrix environment variables are not carried to Bintray deploy phase

### ISSUE THREAD

**Issue by shikloshi** [travis-ci/travis-ci#5185]
I'm building a Go project for windows, linux and darwing using go 1.5+ ```GOOS``` and ```GOARCH``` environment variables.

When I'm trying to use these environment variables inside the ```.travis.yml``` deploy clause they are empty.

When I used ```GIMME_ARCH``` and ```GIMME_OS``` it worked perfectly.

My .travis.yml is:
```
language: go

go:
  - 1.5.1

env:
    matrix:
       - GOOS=windows GOARCH=amd64
       - GOOS=linux GOARCH=amd64
       - GOOS=linux GOARCH=386
       - GOOS=darwin GOARCH=amd64
before_script:
  - go get -d -v ./...

script:
  - go build -v ./...


deploy:
  skip_cleanup: true
  file: bintray_descriptors/${GOOS}_${GOARCH}.json
  provider: bintray
  key: "$API_KEY"
  passphrase: "$PASSPHRASE"
  user: jfrogdev

```

**Comment by BanzaiMan** [comment]
Do you have a build log URL to show this problem?

**Comment by shikloshi** [comment]
Yes: 
https://travis-ci.org/JFrogDev/artifactory-cli-go

**Comment by BanzaiMan** [comment]
It seems to be working, according to this line: https://travis-ci.org/JFrogDev/artifactory-cli-go/jobs/94360884#L217 Your upload is failing, but that is unrelated to the issue under consideration here.

**Comment by shikloshi** [comment]
@BanzaiMan 
This build was working because I changed my .travis.yml completely to work with gimme 0.2.3 (which fixed a bug related to go 1.5.x). 
You were redirected to the latest and successful  build while in other build this bug was happening.

This was the travis.yml:
```
language: go

env:
    matrix:
       - GIMME_OS=windows GIMME_ARCH=amd64
       - GIMME_OS=linux GIMME_ARCH=amd64
       - GIMME_OS=linux GIMME_ARCH=386
       - GIMME_OS=darwin GIMME_ARCH=amd64

before_install:
  - curl -o gimme -sL https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
  - chmod u+x gimme
  - unset GOROOT GOTOOLDIR
  - eval "$(./gimme 1.5)"
  - go version
  - go env

before_script:
  - go get -d -v ./...

script:
  - go build -v ./...

before_deploy:
  - chmod +x ./prepare_bintray_deployment.sh
  - "./prepare_bintray_deployment.sh"

deploy:
  skip_cleanup: true
  file: bintray_descriptors/${GIMME_OS}_${GIMME_ARCH}.json
  provider: bintray
  key: "$API_KEY"
  passphrase: "$PASSPHRASE"
  user: jfrogdev
  on:
    tags: true
```

**Comment by BanzaiMan** [comment]
Could you point to the build that exhibits the problem?
